modules/video_input/video_input.py

```diff
@@ -3,23 +3,72 @@
 """
 
 from .. import image_and_time
-from ..common.modules.camera import camera_device
+from ..common.modules.camera import base_camera
+from ..common.modules.camera import camera_factory
+from ..common.modules.camera import camera_opencv
+from ..common.modules.camera import camera_picamera2
+from ..common.modules.logger import logger
 
 
 class VideoInput:
     """
     Combines image and timestamp together.
     """
 
-    def __init__(self, camera_name: "int | str", save_name: str = "") -> None:
-        self.device = camera_device.CameraDevice(camera_name, 1, save_name)
+    __create_key = object()
 
-    def run(self) -> "tuple[bool, image_and_time.ImageAndTime | None]":
+    @classmethod
+    def create(
+        cls,
+        camera_option: camera_factory.CameraOption,
+        width: int,
+        height: int,
+        config: camera_opencv.ConfigOpenCV | camera_picamera2.ConfigPiCamera2,
+        maybe_image_name: str | None,
+        local_logger: logger.Logger,
+    ) -> "tuple[True, VideoInput] | tuple[False, None]":
+        """
+        camera_option specifies which camera driver to use.
+        width is the width of the images the camera takes in pixels.
+        height is the height of the images the camera takes in pixels.
+        camera_config specifies camera settings.
+        save_prefix is name of iamge log files. Leave as empty string for not logging the images.
+        """
+        result, camera = camera_factory.create_camera(camera_option, width, height, config)
+        if not result:
+            local_logger.error(
+                f"camera factory failed. Current configs were: {camera_option=}, {width=}, {height=}, {config=}. Please try a different set of configs."
```

Comment:
Capitalize. Remove: `Please try ...` , logging should be informational only, no suggestions.